### ISSUE TITLE
yarn upgrade-interactive does not recognize/apply typescript 2.3.4  ❯  2.4.0;  yarn upgrade does

### ISSUE THREAD

**Issue by bentolor** [yarnpkg/yarn#3688]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
`yarn upgrade-interactive` offers "upgrade" of `typescript  2.3.4  ❯  2.3.4.`

Accepting the upgrade has no effect. The next run of `yarn upgrade-interactive` reoffers the same update again.

Deleting the `yarn.lock` file and running `yarn install` again _does have an effect_ and typescript 2.4.0 gets pulled in. Following  `yarn upgrade-interactive`  do no longer offer a new upgrade

**If the current behavior is a bug, please provide the steps to reproduce.**
1. `git checkout git@github.com:bentolor/yarnpkg-demo-upgradebug.git`
2. `yarn upgrade-interactive` – select entry. Press enter:
```
yarn upgrade-interactive                                                                                                            Mi 21 Jun 2017 16:49:25 CEST
yarn upgrade-interactive v0.24.6
? Choose which packages to update. 
? Choose which packages to update. typescript@2.3.4
info Installing "devDependencies"...
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved 1 new dependency.
└─ typescript@2.3.4
Done in 2.96s.
```
3. `yarn upgrade-interactive`  – * here I would expect a empty list`
```
yarn upgrade-interactive                                                                                                   3343ms  Mi 21 Jun 2017 16:49:32 CEST
yarn upgrade-interactive v0.24.6
? Choose which packages to update. (Press <space> to select, <a> to toggle all, <i> to inverse selection)
 devDependencies
❯◯ typescript  2.3.4  ❯  2.3.4.  
```
4. Workaround: 
```
yarn upgrade                                                                                                                23.2s  Mi 21 Jun 2017 16:50:18 CEST
yarn upgrade v0.24.6
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Rebuilding all packages...
success Saved lockfile.
success Saved 1 new dependency.
└─ typescript@2.4.0
Done in 0.99s.
```

**What is the expected behavior?**
`yarn upgrade-interactive` should offer the same upgrade as `yarn upgrade`

**Please mention your node.js, yarn and operating system version.**
````
node --version ; and npm --version ; and yarn --version                                                                    1381ms  Mi 21 Jun 2017 16:50:22 CEST
v6.11.0
3.10.10
0.24.6
````

# Content of the files

## package json
````
{
  "name": "test",
  "version": "0.0.1",
  "private": true,
  "dependencies": {},
  "devDependencies": {
    "typescript": "^2.3.4"
  },
  "engines": {
    "node": ">= 4.0.0",
    "npm": ">= 3.0.0"
  }
}
````

## yarn.lock

````
typescript@^2.3.4:
  version "2.4.0"
  resolved "https://registry.yarnpkg.com/typescript/-/typescript-2.4.0.tgz#aef5a8d404beba36ad339abf079ddddfffba86dd"
````

**Comment by bestander** [comment]
Related to https://github.com/yarnpkg/rfcs/pull/71 maybe

**Comment by BYK** [comment]
This was addressed in 1.0.0 release so closing.
